File: test/cpp/negatives/f4.cpp
#include <iostream>
#include <vector>
using namespace std;

int productOfEvens(const vector<int>& inputList) {
    vector<int> doubled;
    for (int x : inputList) {
        if (x % 2 == 0) {
            doubled.push_back(x * 2);
        }
    }

    int product = 1;
    for (int val : doubled) {
        product *= val;
    }

    return product;
}

int main() {
    vector<int> nums = {1, 2, 3, 4, 5, 6};
    cout << "Product of doubled evens: " << productOfEvens(nums) << endl;
    return 0;
}

File: test/java/negatives/F4.java
import java.util.List;
import java.util.stream.Collectors;

public class F4 {
    public static int productOfEvens(List<Integer> inputList) {
        List<Integer> doubled = inputList.stream()
                .filter(x -> x % 2 == 0)
                .map(x -> x * 2)
                .collect(Collectors.toList());

        return doubled.stream()
                .reduce(1, (a, b) -> a * b);
    }
}
